test/udruntime/include/revalloc.hpp

```diff
@@ -0,0 +1,193 @@
+#include "syscalls.h"
+#include "unistd.h"
+#include <limits>
+#include <memory>
+#include <stdlib.h>
+
+#ifndef __REV_REVALLOC__
+#define __REV_REVALLOC__
+
+/*void* operator new(std::size_t t){
+     void* p = reinterpret_cast<void*>(rev_mmap(0,
+              t,
+              PROT_READ | PROT_WRITE | PROT_EXEC,
+              MAP_PRIVATE | MAP_ANONYMOUS,
+              -1,
+              0));
+    return p;
+}*/
+
+void* mynew( std::size_t t ) {
+  void* p =
+    reinterpret_cast< void* >( rev_mmap( 0,
+                                         t,
+                                         PROT_READ | PROT_WRITE | PROT_EXEC,
+                                         MAP_PRIVATE | MAP_ANONYMOUS,
+                                         -1,
+                                         0 ) );
+  return p;
+}
+
+template< typename T >
+class StandardAllocPolicy {
+public:
+  //    typedefs
+  typedef T                 value_type;
+  typedef value_type*       pointer;
+  typedef const value_type* const_pointer;
+  typedef value_type&       reference;
+  typedef const value_type& const_reference;
+  typedef std::size_t       size_type;
+  typedef std::ptrdiff_t    difference_type;
+
+public:
+  //    convert an StandardAllocPolicy<T> to StandardAllocPolicy<U>
+  template< typename U >
+  struct rebind {
+    typedef StandardAllocPolicy< U > other;
+  };
+
+public:
+  inline explicit StandardAllocPolicy() {
+  }
+
+  inline ~StandardAllocPolicy() {
+  }
+
+  inline explicit StandardAllocPolicy( StandardAllocPolicy const& ) {
+  }
+
+  template< typename U >
+  inline explicit StandardAllocPolicy( StandardAllocPolicy< U > const& ) {
+  }
+
+  //    memory allocation
+  inline pointer
```

Comment:
Just FYI: `inline` is implicit for functions defined inside of classes. But maybe you wanted to be verbose.